### PR TITLE
Support Openshift deployment 

### LINKED ISSUE #69: tornjak-backend ingress error
When using `spire-server.tornjak.ingress.enabled = true` helm wants to create an ingress with

```
spec:
  rules:
    - host: <>
      http:
        paths:
          - path: <>
            pathType: <>
            backend:
              service:
                name: <>
                port:
                  number: tornjak-srv-http
```
See tornjak ingress template: https://github.com/spiffe/helm-charts-hardened/blob/main/charts/spire/charts/spire-server/templates/tornjak-ingress.yaml#L26C1-L27C1
See ingress template in spire-lib: https://github.com/spiffe/helm-charts-hardened/blob/main/charts/spire/templates/_spire-lib.tpl#L125

This results in nginx ingress controller errors, because the number has to be an integer and not a string.

### PR DESCRIPTION
**Pull request by mrsabath**
This PR brings the changes required to support SPIRE charts deployment on OpenShift version 4.13 or higher 

fixes #69

### CODE REVIEW

**Comment by faisal-memon** [comment]
Thanks @mrsabath. It will be great to support OpenShift.

File: examples/production/openshift-values.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      route.openshift.io/termination: "edge"
+    hosts:
+      - host: oidc-discovery.$APP_SUBDOMAIN
+        paths:
+          - path: /
+            pathType: Prefix
+
+tornjak-frontend:
+  apiServerURL: https://spire-tornjak-backend.$APP_SUBDOMAIN/
+  workingDir: /opt/app-root/src
```
Comment (kfox1111) [inline examples/production/openshift-values.yaml:80]: Interesting.... why does this need to change? this a selinux thing? Should we be considering setting this to default everywhere?

File: examples/production/scc-spire-agent.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+fsGroup:
+  type: RunAsAny
+groups: []
+kind: SecurityContextConstraints
+metadata:
+  creationTimestamp: "2023-08-29T18:18:03Z"
+  generation: 11
+  name: scc-spire-agent
+  resourceVersion: "391411"
+  uid: bec548db-1a88-4690-9761-35ffb75464d1
```
Comment (kfox1111) [inline examples/production/scc-spire-agent.yaml:20]: creationTimestamp, generation, resourceVersion and uid shouldn't be in there.

Comment (kfox1111) [reply examples/production/scc-spire-agent.yaml:20]: Same with other objects

File: charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  annotations:
+    kubernetes.io/description: >-
+      {{ $name }} denies access to all host features and requires pods to be
+      run with a UID, and SELinux context that are allocated to the namespace.
+      {{ .Values.restrictedScc.v2 | ternary $v2Description "" }}
+  name: {{ $name }}
+fsGroup:
+  type: MustRunAs
+groups: []
+kind: SecurityContextConstraints
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml:38]: Same with other objects

**Review by kfox1111 — COMMENTED** [review]
Looking good. Thanks for this!

A few things inline

File: charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kfox1111) [reply charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml:38]: (sorry, hit the wrong button and didn't add it to the main review)

File: charts/spire/README.md [hunk 4]
```diff
@@ -117,6 +117,7 @@ Now you can interact with the Spire agent socket from your own application. The
 | upstream-spire-agent.telemetry.prometheus.port | int | `9989` |  |
 | upstream.enabled | bool | `false` | enable upstream agent and driver for use with nested spire. |
 | spiffe-csi-driver.agentSocketPath | string | `"/run/spire/agent-sockets/spire-agent.sock"` | The unix socket path to the spire-agent |
+| spiffe-csi-driver.csiDriverLabels | object | `{}` | Labels to apply to the CSIDriver |
```
Comment (faisal-memon) [inline charts/spire/README.md:120]: Should this just be `labels` to match the other components?

Comment (kfox1111) [reply charts/spire/README.md:120]: They are specifically on the kind: CSIDriver object by the look of it. Where the other labels tend to be on the workload itself? I think its probably the right thing to do if those labels have to be different then the workload labels.

File: charts/spire/Chart.yaml [hunk 5]
```diff
@@ -51,7 +51,7 @@ dependencies:
   - name: tornjak-frontend
     condition: tornjak-frontend.enabled
     repository: file://./charts/tornjak-frontend
-    version: 0.1.0
+    version: 0.1.1
```
Comment (marcofranssen) [inline charts/spire/Chart.yaml:54]:
```suggestion
    version: 0.1.0
```

File: charts/spire/README.md [hunk 6]
```diff
@@ -81,7 +81,7 @@ Now you can interact with the Spire agent socket from your own application. The
 | file://./charts/spire-agent | spire-agent | 0.1.0 |
 | file://./charts/spire-agent | upstream-spire-agent(spire-agent) | 0.1.0 |
 | file://./charts/spire-server | spire-server | 0.1.0 |
-| file://./charts/tornjak-frontend | tornjak-frontend | 0.1.0 |
+| file://./charts/tornjak-frontend | tornjak-frontend | 0.1.1 |
```
Comment (marcofranssen) [inline charts/spire/README.md:84]:
```suggestion
| file://./charts/tornjak-frontend | tornjak-frontend | 0.1.0 |
```

File: charts/spire/charts/spiffe-csi-driver/values.yaml [hunk 7]
```diff
@@ -126,3 +132,12 @@ kubeletPath: /var/lib/kubelet
...
 priorityClassName: ""
+
+restrictedScc:
+  ## @param restrictedScc.enabled Enables the creation of a SecurityContextConstraint based on the restricted SCC with CSI volume support
+  enabled: false
+
+  ## @param restrictedScc.name Set the name of the restricted SCC with CSI support
+  name: ""
+  ## @param restrictedScc.v2 Based up on v2 version of the restricted SCC
+  v2: true
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-csi-driver/values.yaml:143]: Is this param required? Would there be a need to not use the v2?

If there is a need to change this, maybe we should do

```
version: v2
```

to allow for future extensibility.

File: charts/spire/charts/spiffe-oidc-discovery-provider/templates/scc-spire-oidc-discovery-provider.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+readOnlyRootFilesystem: true
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:{{ include "spiffe-oidc-discovery-provider.namespace" . }}:{{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }}
+- system:serviceaccount:{{ include "spiffe-oidc-discovery-provider.namespace" . }}:{{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }}-pre-delete
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-oidc-discovery-provider/templates/scc-spire-oidc-discovery-provider.yaml:15]: To align with our way of formatting the yamls.

```suggestion
  - system:serviceaccount:{{ include "spiffe-oidc-discovery-provider.namespace" . }}:{{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }}
  - system:serviceaccount:{{ include "spiffe-oidc-discovery-provider.namespace" . }}:{{ include "spiffe-oidc-discovery-provider.serviceAccountName" . }}-pre-delete
```

File: charts/spire/charts/spire-agent/templates/scc-spire-agent.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:{{ include "spire-agent.namespace" . }}:{{ include "spire-agent.serviceAccountName" . }}
+volumes:
+- configMap
+- hostPath
+- projected
+- secret
```
Comment (marcofranssen) [inline charts/spire/charts/spire-agent/templates/scc-spire-agent.yaml:19]:
```suggestion
  - system:serviceaccount:{{ include "spire-agent.namespace" . }}:{{ include "spire-agent.serviceAccountName" . }}
volumes:
  - configMap
  - hostPath
  - projected
  - secret
```

File: charts/spire/charts/spire-server/templates/tornjak-ingress.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: {{ include "spire-server.namespace" . }}
+  labels:
+    {{ include "spire-server.labels" . | nindent 4}}
+  {{- with .Values.tornjak.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.ingress.servicePort "global" .Values.global) | nindent 2 }}
+{{- end }}
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/templates/tornjak-ingress.yaml:16]: This seems to be unrelated to the openshift changes?

Can we move this into a separate PR so it also shows up nicely in the relase notes as "Add Tornjak ingress".

File: charts/spire/charts/spire-server/values.yaml [hunk 11]
```diff
@@ -562,6 +562,31 @@ tornjak:
...
+      - host: tornjak-backend.example.org
+        paths:
+          - path: /
+            pathType: Prefix
+    ## @param tornjak.ingress.tls [array] Secrets containing TLS certs to enable https on ingress
+    tls: []
+    #  - secretName: chart-example-tls
+    #    hosts:
+    #      - tornjak-backend.example.org
+
```
Comment (marcofranssen) [inline charts/spire/charts/spire-server/values.yaml:589]: See previous.

File: charts/spire/charts/tornjak-frontend/Chart.yaml [hunk 12]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: tornjak-frontend
 description: A Helm chart to deploy Tornjak frontend
 type: application
-version: 0.1.0
+version: 0.1.1
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/Chart.yaml:5]:
```suggestion
version: 0.1.0
```

File: charts/spire/charts/tornjak-frontend/README.md [hunk 13]
```diff
@@ -2,7 +2,7 @@
 
 <!-- This README.md is generated. Please edit README.md.gotmpl -->
 
-![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.2.2](https://img.shields.io/badge/AppVersion-v1.2.2-informational?style=flat-square)
+![Version: 0.1.1](https://img.shields.io/badge/Version-0.1.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.2.2](https://img.shields.io/badge/AppVersion-v1.2.2-informational?style=flat-square)
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/README.md:5]:
```suggestion
![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.2.2](https://img.shields.io/badge/AppVersion-v1.2.2-informational?style=flat-square)
```

File: charts/spire/charts/tornjak-frontend/README.md [hunk 14]
```diff
@@ -52,6 +52,7 @@ port forwarding. See the chart NOTES output for more details.
 
 * <https://github.com/spiffe/tornjak>
 
+
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/README.md:55]:
```suggestion
```

File: charts/spire/charts/tornjak-frontend/templates/deployment.yaml [hunk 15]
```diff
@@ -53,9 +53,9 @@ spec:
           {{- end }}
           volumeMounts:
             - name: cache
-              mountPath: /usr/src/app/node_modules/.cache
+              mountPath: {{ .Values.workingDir }}/node_modules/.cache
             - name: env
-              mountPath: /usr/src/app/build/tmp
+              mountPath: {{ .Values.workingDir }}/build/tmp
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/templates/deployment.yaml:58]: Maybe default this to a path that works for Openshift, so we don't need that configurable.

File: charts/spire/charts/tornjak-frontend/templates/ingress.yaml [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: {{ include "tornjak-frontend.namespace" . }}
+  labels:
+    {{ include "tornjak-frontend.labels" . | nindent 4}}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.ingress "svcName" $fullName "port" .Values.service.port "global" .Values.global) | nindent 2 }}
+{{- end }}
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/templates/ingress.yaml:16]: Isolate into separate PR, see also comments before.

File: charts/spire/charts/tornjak-frontend/values.yaml [hunk 17]
```diff
@@ -113,6 +113,32 @@ startupProbe:
   failureThreshold: 6
   successThreshold: 1
 
+## @param workingDir Path containing the Tornjak frontend within the image
+workingDir: /usr/src/app
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/values.yaml:117]: See earlier comment, we might just default this to something that works for openshift so the configurable isn't required.

File: charts/spire/charts/tornjak-frontend/values.yaml [hunk 18]
```diff
@@ -113,6 +113,32 @@ startupProbe:
...
+  hosts:
+    - host: tornjak-frontend.example.org
+      paths:
+        - path: /
+          pathType: Prefix
+  ## @param ingress.tls [array] Secrets containing TLS certs to enable https on ingress
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - tornjak-frontend.example.org
```
Comment (marcofranssen) [inline charts/spire/charts/tornjak-frontend/values.yaml:140]: See other comments on moving this to separate PR.

File: examples/production/openshift-values.yaml [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+
+spire-agent:
+  podSecurityContext:
+    runAsUser: null
+    runAsGroup: null
+    fsGroup: null
+
+upstream-spire-agent:
+  podSecurityContext:
+    runAsUser: null
```
[the remark this answers is not in the document]
Comment (marcofranssen) [reply examples/production/openshift-values.yaml:54]: Maybe `podSecurityContext: {}` is sufficient as well?

I assume this is done so the Openshift SSC handles these things now.

File: examples/production/scc-spire-agent.yaml [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:spire-system:spire-agent
+volumes:
+- configMap
+- hostPath
+- projected
+- secret
```
Comment (marcofranssen) [inline examples/production/scc-spire-agent.yaml:32]:
```suggestion
  - system:serviceaccount:spire-system:spire-agent
volumes:
  - configMap
  - hostPath
  - projected
  - secret
```

File: examples/production/scc-spire-spiffe-csi-driver.yaml [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:spire-system:spire-spiffe-csi-driver
+volumes:
+- configmap
+- hostPath
+- secret
```
Comment (marcofranssen) [inline examples/production/scc-spire-spiffe-csi-driver.yaml:31]:
```suggestion
  - system:serviceaccount:spire-system:spire-spiffe-csi-driver
volumes:
  - configmap
  - hostPath
  - secret
```

**Review by marcofranssen — CHANGES_REQUESTED** [review]
@mrsabath Great work on this one. Few inline suggestions and questions.

Can we also introduce the ingress in a separate PR so we can merge that first and have it captured in release notes? Also see the inline feedback on those occurences.

File: charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+seLinuxContext:
+  type: MustRunAs
+readOnlyRootFilesystem: false
+metadata:
+  annotations:
+    kubernetes.io/description: >-
+      {{ $name }} denies access to all host features and requires pods to be
+      run with a UID, and SELinux context that are allocated to the namespace.
+      {{ .Values.restrictedScc.v2 | ternary $v2Description "" }}
+  name: {{ $name }}
```
Comment (kfox1111) [inline charts/spire/charts/spiffe-csi-driver/templates/scc-restricted-csi.yaml:36]: metadata.name being near the top in the standard ordering really helps us users when we are looking at what everything does.

File: charts/spire/charts/tornjak-frontend/templates/deployment.yaml (hunk 15 above)
Comment (kfox1111) [reply charts/spire/charts/tornjak-frontend/templates/deployment.yaml:58]: Yeah, I'd like to know a bit more why the change was needed. If the reasoning is compatible with hard coding it to the new location, that would be best. But it may imply other clusters may need it configured other ways, in which case it should be configurable. So, knowing why openshift needed this change is important.

**Review by kfox1111 — COMMENTED** [review]
Looking better. Some comments inline

File: charts/spire/charts/spiffe-csi-driver/templates/scc-spiffe-csi-driver.yaml [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:{{ include "spiffe-csi-driver.namespace" . }}:{{ include "spiffe-csi-driver.serviceAccountName" . }}
+volumes:
+- configmap
+- hostPath
+- secret
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-csi-driver/templates/scc-spiffe-csi-driver.yaml:18]:
```suggestion
  - configmap
  - hostPath
  - secret
```

File: charts/spire/charts/spiffe-csi-driver/templates/scc-spiffe-csi-driver.yaml [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+  name: {{ include "spiffe-csi-driver.fullname" . }}
+readOnlyRootFilesystem: true
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: RunAsAny
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:{{ include "spiffe-csi-driver.namespace" . }}:{{ include "spiffe-csi-driver.serviceAccountName" . }}
```
Comment (marcofranssen) [inline charts/spire/charts/spiffe-csi-driver/templates/scc-spiffe-csi-driver.yaml:14]:
```suggestion
  - system:serviceaccount:{{ include "spiffe-csi-driver.namespace" . }}:{{ include "spiffe-csi-driver.serviceAccountName" . }}
```

File: charts/spire/README.md [hunk 25]
```diff
@@ -494,7 +514,9 @@ Now you can interact with the Spire agent socket from your own application. The
 | tornjak-frontend.startupProbe.timeoutSeconds | int | `5` | Timeout seconds for startupProbe |
 | tornjak-frontend.tolerations | list | `[]` |  |
 | tornjak-frontend.topologySpreadConstraints | list | `[]` |  |
+| tornjak-frontend.workingDir | string | `"/usr/src/app"` | Path containing the Tornjak frontend within the image |
```
[the remark this answers is not in the document]
Comment (sabre1041) [reply charts/spire/README.md]: users can provide their own image and the contents of the working directory may be alternate to the default value

File: charts/spire/charts/tornjak-frontend/values.yaml (hunk 17 above)
Comment (sabre1041) [reply charts/spire/charts/tornjak-frontend/values.yaml:117]: This is to support arbitrary UID's that are part of OpenShift. The default image does not include support and if an alternate image is used, it may not match the default value and this directory may not be present

File: examples/production/openshift-values.yaml (hunk 1 above)
Comment (sabre1041) [reply examples/production/openshift-values.yaml:80]: arbitrary UID support

File: charts/spire/charts/spiffe-csi-driver/values.yaml (hunk 7 above)
Comment (sabre1041) [reply charts/spire/charts/spiffe-csi-driver/values.yaml:143]: This has to do with SCC versioning and specific to OpenShift

File: examples/production/openshift-values.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (sabre1041) [reply examples/production/openshift-values.yaml:54]: There have been challenges working with overriding nested values. 3.13 provided solutions, but this is mainly a safety net to support multiple versions

[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/production/openshift-values.yaml:54]: The defaults are unset though. So your clearing something unset? Still not sure I understand.

File: charts/spire/charts/spiffe-csi-driver/values.yaml (hunk 7 above)
Comment (mrsabath) [reply charts/spire/charts/spiffe-csi-driver/values.yaml:143]: We can address @marcofranssen suggestion to make it more expandable in the future.

File: examples/production/openshift-values.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/production/openshift-values.yaml:54]: Are there a set of uids that would be more suitable? Unrestricting them is less preferable then hardening it?

[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/production/openshift-values.yaml:54]: Still curious on this one. This is removing the user runtime setup? So its running as root again? There better options we could pick?

File: charts/spire/charts/spire-server/templates/tornjak-ingress.yaml [hunk 26]
```diff
@@ -13,8 +13,8 @@ metadata:
   {{- end }}
 spec:
   {{- if eq (include "spire-tornjak.connectionType" .) "http" }}
-  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" "tornjak-srv-http") | nindent 2 }}
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.service.ports.http) | nindent 2 }}
   {{- else -}}
-  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" "tornjak-srv-https") | nindent 2 }}
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.service.ports.https) | nindent 2 }}
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/templates/tornjak-ingress.yaml:18]: This is from a previous version I think.

Comment (kfox1111) [reply charts/spire/charts/spire-server/templates/tornjak-ingress.yaml:28]: After comparing again with main, looks ok now. Thanks.

File: examples/production/openshift-values.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (sabre1041) [reply examples/production/openshift-values.yaml:54]: OpenShift does not allow that by default and will assign a user ID based on the SecurityContextConstraint. Given this values file is specific to OpenShift, we are complying with the rules/configuration within the platform

[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/production/openshift-values.yaml:58]: It doesn't seem to be specifying one in the SecurityContextConstraint, so is it running as root then? How do we harden it so it runs as a restricted user instead?

File: examples/production/scc-spire-agent.yaml [hunk 27]
```diff
@@ -0,0 +1,32 @@
+apiVersion: security.openshift.io/v1
```
Comment (kfox1111) [inline examples/production/scc-spire-agent.yaml:1]: Is this file redundant because of charts/spire/charts/spire-agent/templates/scc-spire-agent.yaml?

File: examples/production/scc-spire-spiffe-csi-driver.yaml [hunk 28]
```diff
@@ -0,0 +1,31 @@
+apiVersion: security.openshift.io/v1
```
Comment (kfox1111) [inline examples/production/scc-spire-spiffe-csi-driver.yaml:1]: Is this file redundant because of Is this file redundant because of charts/spire/charts/spiffe-csi-driver/templates/scc-spire-spiffe-csi-driver.yaml?

File: charts/spire/charts/spire-server/templates/tornjak-ingress.yaml [hunk 29]
```diff
@@ -23,8 +23,8 @@ metadata:
   {{- end }}
 spec:
   {{- if eq (include "spire-tornjak.connectionType" .) "http" }}
-  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" "tornjak-srv-http" "Values" .Values) | nindent 2 }}
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.service.ports.http) | nindent 2 }}
   {{- else -}}
-  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" "tornjak-srv-https" "Values" .Values) | nindent 2 }}
+  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.service.ports.https) | nindent 2 }}
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/templates/tornjak-ingress.yaml:28]:
```suggestion
  {{ include "spire-lib.ingress-spec" (dict "ingress" .Values.tornjak.ingress "svcName" (include "spire-tornjak.servicename" .) "port" .Values.tornjak.service.ports.https "Values" .Values) | nindent 2 }}
```

File: examples/openshift/openshift-values.yaml [hunk 30]
```diff
@@ -0,0 +1,94 @@
...
+
+spire-server:
+  tornjak:
+    image:
+      registry: ghcr.io
+      repository: spiffe/tornjak-backend
+      tag: ubi-v1.4.1
+    ingress:
+      annotations:
+        route.openshift.io/termination: "edge"
```
Comment (kfox1111) [inline examples/openshift/openshift-values.yaml:15]:
```suggestion
```

File: examples/openshift/openshift-values.yaml [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+    k8sbundle:
+      namespace: spire-system
+  podSecurityContext:
+    # These are unset so that openshift can automatically assign its own restricted uids to the pods
+    runAsUser: null
+    runAsGroup: null
+    fsGroup: null
+  ingress:
+    annotations:
+      route.openshift.io/termination: "passthrough"
```
Comment (kfox1111) [inline examples/openshift/openshift-values.yaml:30]:
```suggestion
```

File: examples/openshift/openshift-values.yaml [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+    fsGroup: null
+
+spiffe-oidc-discovery-provider:
+  podSecurityContext:
+    runAsUser: null
+    runAsGroup: null
+    fsGroup: null
+  ingress:
+    annotations:
+      route.openshift.io/termination: "edge"
```
Comment (kfox1111) [inline examples/openshift/openshift-values.yaml:57]:
```suggestion
```

File: examples/openshift/openshift-values.yaml [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+
+tornjak-frontend:
+  workingDir: /opt/app-root/src
+  image:
+    registry: ghcr.io
+    repository: spiffe/tornjak-frontend
+    tag: ubi-v1.4.1
+  ingress:
+    annotations:
+      route.openshift.io/termination: "edge"
```
Comment (kfox1111) [inline examples/openshift/openshift-values.yaml:67]:
```suggestion
```

**Review by kfox1111 — COMMENTED** [review]
With the removal of the ingress annotations from the openshift-values.yaml as suggested, I approve this pr.

The non ingress parts work. Tested on openshift local. Before the other pr merges, it should be possible to re-add the right ingress settings manually via values.

The ingress bits can get added back in automatically with the ingress controller type openshift patch. So, lets work on the last bits there?

**Review by kfox1111 — APPROVED** [review]

File: project/openshift.md [hunk 34]
```diff
@@ -0,0 +1,37 @@
+# OpenShift notes for K8S developers
+
+## SecurityContexts
+
+OpenShift automatically generates uid/gid's for pods. They should not be set to get this behavior.
+
+## CSIDriver issues
+
+A workload in a restricted namespace can not access a csidriver that isn't labeled:
+```yaml
```
Comment (marcofranssen) [inline project/openshift.md:10]:
```suggestion

```yaml
```

File: project/openshift.md [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+The defaults though are too chatty. It puts audit/warn still at restricted.
+
+## Ingress
+
+Ingress objects automatically create Role objects in the same namespace, when the ingress object is viewed as valid by openshift, if not it is ignored. A missing Role object is a sure sign that something is wrong in the Ingress.
+
+Some things to watch out for.
+
+When the ingress is annotated:
+```yaml
```
Comment (marcofranssen) [inline project/openshift.md:29]:
```suggestion

```yaml
```

File: project/openshift.md [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+  "route.openshift.io/termination": "passthrough" 
+```
+
+The ingress object can not have a path specified and the pathType needs to be ImplementationSpecific
+
+
```
Comment (marcofranssen) [inline project/openshift.md:35]:
```suggestion
```

**Review by marcofranssen — APPROVED** [review]
LGTM :rocket:

Thanks for addressing the feedback.
